lua/starfall/editor/docs.lua

```diff
@@ -161,54 +237,81 @@ local function parse(parsing, data)
 	end
 end
 
-local function scan(src, realm)
-	local linetbl = string.Explode("\r?\n", src, true)
-	local i = 0
-	local function lines() i=i+1 return linetbl[i] end
+local function get_lines(self)
+    local result = {}
+    local from, line_n = 1, 0
+    local delim_from, delim_to = string_find( self, "\r?\n", from  )
+    while delim_from do
+      local str = string_sub( self, from , delim_from-1 )
+      from = delim_to + 1
+      line_n = line_n + 1
+      result[line_n] = str
+      delim_from, delim_to = string_find( self, "\r?\n", from  )
+    end
+    result[line_n+1] = string_sub( self, from  )
+    return result
+end
+--- Scan function
+-- @param string src Source code
+-- @param string file_name Source file name.
+-- @param boolean type_only Whether to only parse types / classes.
+local function scan(src, realm, type_only)
+	-- https://github.com/thegrb93/StarfallEx/blob/master/lua/starfall/...
+	local file_path = string_match(curfile, "%.%./lua/starfall/(libs_.+/.*)") -- libs_sh/... path that will be used for links with [src] on the sfhelper to the github.
 	local parsing
-	for line in lines do
+
+	for line_n, line in next, get_lines(src) do
 		if parsing then
-			local data = string.match(line, "^%s*%-%-%-*(.*)")
+			local data = string_match(line, "^%s*%-%-%-*(.*)")
 			if data then
-				parse(parsing, data)
+				parse(parsing, data, type_only, line_n)
 			else
-				while line and not string.find(line, "%S") do -- Find next non-empty line
```

Comment:
This was there in case some empty lines were between the doc and the function definition. You're sure its not needed?